core/src/main/java/org/apache/iceberg/BaseMetastoreCatalog.java

```diff
@@ -110,9 +110,17 @@ public Transaction newReplaceTableTransaction(
       throw new NoSuchTableException("No such table: " + identifier);
     }
 
-    String baseLocation = location != null ? location : defaultWarehouseLocation(identifier);
     Map<String, String> tableProperties = properties != null ? properties : Maps.newHashMap();
-    TableMetadata metadata = TableMetadata.newTableMetadata(schema, spec, baseLocation, tableProperties);
+
+    TableMetadata metadata;
+    if (ops.current() != null) {
```

Comment:
I'm not clear on whether this really should be the right behavior.  Basically we're saying that a replace table will keep the existing location (as opposed to using defaults).  I suspect we don't have create or replace with location semantics, but this is making some assumptions that a replacement is somehow the same as the old.  If we were to go with id based pathing convention, this wouldn't work.

I don't think this is an issue at this point, but it might make sense to push this down to the location provider.

Reply:
I think this is correct because the `REPLACE TABLE` doesn't completely wipe out the old table. In most ways, it is the same table.

Calling `buildReplacement` will replace the schema and partition spec so that the transaction can add a new snapshot. Table history, old snapshots, and existing table properties are kept so that you can inspect the table and don't need to add table configuration every time you run the SQL.

We could add a flag to turn off this behavior and wipe out the old by default, but I don't think that's what users really want. It makes sense for things like default format and other settings to persist across replace operations, so that table configuration and table operations are orthogonal.

Reply:
ok, I guess that makes sense.  I assumed the semantics to be the same a transactional `drop` and `create`, but based on a little searching, it's much less clear than that.  For example, with db2 create or replace will actually retain all data (assuming if aligns with the new table definition) by default.

I agree that preserving table properties make sense, but wouldn't this wipe out comments (which we may also want to preserve)

Reply:
It would probably wipe out column comments. I'll have to fix that.